server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java

```diff
@@ -191,36 +189,40 @@ public Invocations getInvocations() {
-               cacheInfo.type = cacheConfiguration.clustering().cacheMode().toCacheType();
-               cacheInfo.status = cache.getStatus().name();
-               cacheInfo.size = cache.size();
-               cacheInfo.simpleCache = cacheConfiguration.simpleCache();
-               cacheInfo.transactional = cacheConfiguration.transaction().transactionMode().isTransactional();
-               cacheInfo.persistent = cacheConfiguration.persistence().usingStores();
-               cacheInfo.bounded = cacheConfiguration.expiration().maxIdle() != -1 ||
-                     cacheConfiguration.expiration().lifespan() != -1;
-               cacheInfo.secured = cacheConfiguration.security().authorization().enabled();
-               cacheInfo.indexed = cacheConfiguration.indexing().index().isEnabled();
-               cacheInfo.hasRemoteBackup = cacheConfiguration.sites().hasEnabledBackups();
-               return cacheInfo;
-            }).collect(Collectors.toSet());
-
-            byte[] bytes = objectMapper.writeValueAsBytes(caches);
-            responseBuilder.contentType(APPLICATION_JSON).entity(bytes);
-
-         } catch (JsonProcessingException e) {
-            responseBuilder.status(HttpResponseStatus.INTERNAL_SERVER_ERROR);
-         }
-
-         return responseBuilder.build();
-      }, executor);
+      return Flowable.fromIterable(cacheManager.getCacheNames())
+            .map(cacheManager::getCache)
+            .flatMap(cache ->
+               RxJavaInterop.<Long>completionStageToPublisher()
+                     .apply(cache.sizeAsync())
+                     .map(size -> {
+                        CacheInfo cacheInfo = new CacheInfo();
+                        cacheInfo.name = cache.getName();
+                        Configuration cacheConfiguration = cache.getCacheConfiguration();
+                        cacheInfo.type = cacheConfiguration.clustering().cacheMode().toCacheType();
+                        cacheInfo.status = cache.getStatus().name();
+                        cacheInfo.size = size;
+                        cacheInfo.simpleCache = cacheConfiguration.simpleCache();
+                        cacheInfo.transactional = cacheConfiguration.transaction().transactionMode().isTransactional();
+                        cacheInfo.persistent = cacheConfiguration.persistence().usingStores();
+                        cacheInfo.bounded = cacheConfiguration.expiration().maxIdle() != -1 ||
+                              cacheConfiguration.expiration().lifespan() != -1;
+                        cacheInfo.secured = cacheConfiguration.security().authorization().enabled();
+                        cacheInfo.indexed = cacheConfiguration.indexing().index().isEnabled();
+                        cacheInfo.hasRemoteBackup = cacheConfiguration.sites().hasEnabledBackups();
+                        return cacheInfo;
+                        // Only request 10 caches sizes in parallel
+                     }), 10)
```

Comment:
Any rationale behind 10?

Reply:
Just wanted to limit it to something lower. RxJava2 would use 128 by default and that seems like way too many. To be honest I am debating if we want to lower it to 1 even.

Reply:
+1 to use limit it to 1

Reply:
I have created https://github.com/infinispan/infinispan/pull/7568